src/mono/wasm/runtime/library_mono.js

```diff
@@ -2252,6 +2252,16 @@ var MonoSupportLib = {
 		};
 		debugger;
 	},
+
+	mono_wasm_add_files: function (assembly_ptr, assemnly_len, pdb_ptr, pdb_len) {
+		const assembly_data = new Uint8Array(Module.HEAPU8.buffer, assembly_ptr, assemnly_len);
+		const pdb_data = pdb_ptr ? new Uint8Array(Module.HEAPU8.buffer, pdb_ptr, pdb_len) : null;
+		MONO.mono_wasm_raise_debug_event({
+			eventName: 'ADD_ASSEMBLY_PDB',
+			assembly_data: Array.from(assembly_data),
```

Comment:
This returns an array, which gets stringified, so `[4,12,2, ... ]`. It would be better to return a base64 string instead.
I was trying ` const assembly_b64 = btoa (String.fromCharCode (...assembly_data));`, but that ends up exceeding the call stack size. We could encode that manually in a loop, but maybe you know of some better way to do this?

The good way to do this would be to use cdp's IO api, but that might be too much new code close to rc2. I will try that locally anyway.

Reply:
You can add this to `$MONO` object - https://gist.github.com/radical/00c86f5617fb8f26db5a3b6b5ff58361 . It's based on JSIL code, thanks to @kg .

And use it like:
```
		const assembly_data = new Uint8Array(Module.HEAPU8.buffer, assembly_ptr, assemnly_len);
		const assembly_b64 = MONO._base64Converter.toBase64StringImpl(assembly_data);

		let pdb_b64;
		if (pdb_ptr) {
			const pdb_data = new Uint8Array(Module.HEAPU8.buffer, pdb_ptr, pdb_len);
			pdb_b64 = MONO._base64Converter.toBase64StringImpl(pdb_data);
		}
```

I tried it out, and it works fine. Once that is on here, we can ask her to review it.

Reply:
Yeah! I spent a bit of time playing with different approaches here but wasn't able to find something that worked or wasn't unbearably slow. I'll integrate these changes and see how that goes.

Reply:
The change has been integrated. I see this is a code change pulled out from another codebase. Do we need to add additional tests for it here?

Also: does it make sense to put this under the BINDING namespace?